### ISSUE TITLE
[api.cal.com] Get Bookings, Filter By Upcoming

### ISSUE THREAD

**Issue by hunterwilhelm** [calcom/cal.com#15983]
# [api.cal.com] Get Bookings, Filter By Upcoming
### My Problem

The endpoint in mention: https://cal.com/docs/enterprise-features/api/api-reference/bookings

I'm looking for a way to get only upcoming bookings, if I request for all live bookings every hour on a cron job, it would eventually slow down, due to the sheer number of bookings we are planning to have. 

For example in 1 year, the endpoint would be returning 9600 bookings, which would be about 7.08 megabytes of information per hour. If cal.com is being charged per read, that might be undesirable as well.

### Possible Solution

I looked at the internal API and it shows this type of query is possible.

https://app.cal.com/api/trpc/bookings/get?batch=1&input=%7B%220%22%3A%7B%22json%22%3A%7B%22limit%22%3A10%2C%22filters%22%3A%7B%22status%22%3A%22past%22%7D%2C%22cursor%22%3A19%7D%7D%7D

Decoded TRPC url:

https://app.cal.com/api/trpc/bookings/get?batch=1&input={"0":{"json":{"limit":10,"filters":{"status":"past"},"cursor":19}}}

Notice that it has `"filters":{"status":"past"}`. Could we expose this to the API?

### Alternatives I've considered

Pagination doesn't seem to be a solution since it doesn't give me a way to see how many results there are. Therefore, I don't know what page to jump to quickly. The most recent results are the last pages of the query, which feels backwards, but it's fine.

### Additional context

Other valid solutions: 
* Showing the total number of pages or bookings in the bookings endpoint
* Allow sorting by most recent first
* Allow filtering by upcoming 

### Requirement/Document

https://cal.com/docs/enterprise-features/api/api-reference/bookings#pagination

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.
  - Follow Best Practices lined out in our [Contributor Docs](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)

**Comment by hunterwilhelm** [comment]
Small update: I found a way to sort by createdAt and modifiedAt, but that doesn't help me since someone could have scheduled far into the future, and possibly it could be on a different page entirely if I tried to combine this technique with pagination.

I looked in the code and found this:

https://github.com/calcom/cal.com/blob/27d331ad2df218f0fc36e0d0dbdf372c232ebf1c/apps/api/v1/pages/api/bookings/_get.ts

It shows we can do sort by createdAt or modifiedAt. For example, this works.

https://api.cal.com/v1/bookings?sortBy=createdAt&order=desc

Even though this feature isn't in the public docs, I'll hold off creating an issue for this until this request is decided.
